x/wasm-storage/types/wasm.go

```diff
@@ -0,0 +1,51 @@
+package types
+
+import (
+	fmt "fmt"
+	"strings"
+
+	"github.com/hyperledger/burrow/crypto"
+)
+
+// MaxWasmSize is the maximum size of Wasm bytecode.
+const MaxWasmSize = 800 * 1024
+
+func validateWasmCode(s []byte) error {
+	if len(s) == 0 {
+		return fmt.Errorf("empty Wasm code")
+	}
+	if len(s) > MaxWasmSize {
+		return fmt.Errorf("Wasm code cannot be longer than %d bytes", MaxWasmSize)
+	}
+	return nil
+}
+
+// NewWasm constructs a new Wasm object given bytecode and Wasm type.
+// It panics if it fails to compute hash of bytecode.
+func NewWasm(bytecode []byte, wasmType WasmType) *Wasm {
+	hash := crypto.Keccak256(bytecode)
+	if hash == nil {
+		panic("failed to compute hash")
+	}
+
+	return &Wasm{
+		Hash:     hash,
+		Bytecode: bytecode,
```

Comment:
Maybe for future usage we can add the block it was added to? So we don't have to migrate our state if we ever introduce pruning

Reply:
I agree. Not sure if we're going with block number-based vs block time-based pruning but I'll add block number for now.

Reply:
Acutally I added timestamp of when the object was added because I think pruning should be based on timestamps (not important anyways right now)

Reply:
To be honest, i think block number is probably better because it's more consistent. Timestamps are not really to be trusted. But maybe it doesn't matter in this use case. We can resolve this one for now.